applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_law.h

```diff
@@ -36,26 +36,24 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoThermalDispersionLaw : public Con
 
     explicit GeoThermalDispersionLaw(SizeType NumberOfDimensions);
 
-    ConstitutiveLaw::Pointer Clone() const override;
+    ~GeoThermalDispersionLaw() override = default;
 
-    SizeType WorkingSpaceDimension() override;
+    GeoThermalLaw::Pointer Clone() const override;
 
-    Matrix CalculateThermalDispersionMatrix(const Properties& rProp, const ProcessInfo& rProcessInfo) const;
+    Matrix CalculateThermalDispersionMatrix(const Properties& rProp, const ProcessInfo& rProcessInfo) const override;
 
 private:
-    SizeType mNumberOfDimensions;
-
     friend class Serializer;
 
     void save(Serializer& rSerializer) const override
     {
-        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, ConstitutiveLaw)
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, GeoThermalLaw)
         rSerializer.save("NumberOfDimensions", mNumberOfDimensions);
     }
 
     void load(Serializer& rSerializer) override
     {
-        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, ConstitutiveLaw)
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, GeoThermalLaw)
         rSerializer.load("NumberOfDimensions", mNumberOfDimensions);
```

Comment:
I think we're now saving and loading the mNumberOfDimensions twice (once in the base and once in the derived classes). I think it would be best to not have a protected member mNumberOfDimensions, but just have one in both the derived classes instead. Protected variables are counted as code smells.